Source/WebCore/dom/Microtasks.h

```diff
@@ -34,6 +34,31 @@ class VM;
 
 namespace WebCore {
 
+class MicrotaskCheckpointScope final {
+    WTF_FORBID_HEAP_ALLOCATION;
+    WTF_MAKE_NONCOPYABLE(MicrotaskCheckpointScope);
+    WTF_MAKE_NONMOVABLE(MicrotaskCheckpointScope);
+public:
+    explicit MicrotaskCheckpointScope(EventLoop& eventLoop)
+    {
+        Ref protectedEventLoop = eventLoop;
+        protectedEventLoop->forEachAssociatedContext([this](auto& context) {
+            m_savedNestingLevels.set(context, context.timerNestingLevel());
+            context.setTimerNestingLevel(0);
+        });
+    }
+
+    ~MicrotaskCheckpointScope()
+    {
+        for (auto contextAndLevel : m_savedNestingLevels) {
```

Comment:
Do: `auto [key, value] : m_savedNestingLevels `.
Then you can just do: `Ref { key }->setTimerNestingLevel(contextAndLevel.value)`.